filebeat/channel/runner_mock_test.go

```diff
@@ -69,17 +70,19 @@ func (r runnerFactoryMock) Assert(t *testing.T) {
 	// we need to make sure `Assert` is called after `Create`
 	require.Len(t, r.cfgs, r.clientCount)
 
+	sameBacking := func(a, b any) bool {
+		return reflect.ValueOf(a).UnsafePointer() == reflect.ValueOf(b).UnsafePointer()
+	}
+
 	t.Run("new processing configuration each time", func(t *testing.T) {
 		for i, c1 := range r.cfgs {
 			for j, c2 := range r.cfgs {
 				if i == j {
 					continue
 				}
 
-				require.NotSamef(t, c1.Processing, c2.Processing, "processing configuration cannot be re-used")
-				require.NotSamef(t, c1.Processing.Meta, c2.Processing.Meta, "`Processing.Meta` cannot be re-used")
-				require.NotSamef(t, c1.Processing.Fields, c2.Processing.Fields, "`Processing.Fields` cannot be re-used")
-				require.NotSamef(t, c1.Processing.Processor, c2.Processing.Processor, "`Processing.Processor` cannot be re-used")
+				require.Falsef(t, sameBacking(c1.Processing.Meta, c2.Processing.Meta), "`Processing.Meta` cannot be re-used")
```

Comment:
I think we accomplish the same thing (without needing unsafe pointer) by passing map pointers into `NotSame` . e.g.

```
				require.NotSamef(t, &c1.Processing.Meta, &c2.Processing.Meta, "`Processing.Meta` cannot be re-used")
				require.NotSamef(t, &c1.Processing.Fields, &c2.Processing.Fields, "`Processing.Fields` cannot be re-used")
```

Reply:
I might be missing something, but I think it does not. I think the addresses of the maps are not the addresses of the backing storage of said map, I'm not super versed in go so I might be misinterpreting it:
```
	x := make(map[string]int)
	x["foo"] = 2
	y := x
	y["bar"] = 3

	require.NotSame(t, &x, &y, "lalalal") <--- should blow up here, but doesn't
	fmt.Printf("same backing = %v\n", sameBacking(x, y))
	require.Falsef(t, sameBacking(x, y), "`Processing.Meta` cannot be re-used") <--- blows up here
```
Here sameBacking(x, y) is true, but require.NotSame(&x, &y) is also true (so it passes when it shouldn't)

Reply:
You are right. This needs the underlying map data to be correct.